src/workerd/io/actor-cache.h

```diff
@@ -319,23 +319,10 @@ class ActorCache final: public ActorCacheInterface {
     // The value was set by the app via put() or delete(), and we have not yet initiated a write
     // to disk. The entry is appended to `dirtyList` whenever entering this state.
     //
-    // Next state: FLUSHING (if we begin flushing to disk) or NOT_IN_CACHE (if a new put()/delete()
+    // Next state: CLEAN (if the flush succeeds) or NOT_IN_CACHE (if a new put()/delete()
     // overwrites this entry first).
     DIRTY,
 
-    // The value is dirty but an RPC is in-flight to write this value to disk. If the value is
-    // overwritten in the meantime, or the write fails, then the state needs to change back to
-    // DIRTY. If the write completes successfully and the state is still FLUSHING, then the state
-    // can progress to CLEAN.
-    //
-    // The entry remains in `dirtyList` while in this state. Since newly-dirty entries are always
-    // appended to `dirtyList`, it's guaranteed that all `FLUSHING` entries in `dirtyList` come
-    // before all `DIRTY` entries.
-    //
-    // Next state: CLEAN (if the flush completes) or NOT_IN_CACHE (if a new put()/delete()
-    // overwrites the entry first).
-    FLUSHING,
```

Comment:
Do you understand the motivation for this committ, @MellowYarker ? The commit message gives no explanation of why removing the `FLUSHING` state makes entry removal easier or why the `FLUSHING` state isn't needed anymore, and I don't understand why we'd make this kind of substantive change without a clear reason -- it's the sort of thing that leads to nasty regressions if not very carefully reviewed.

Reply:
And as a TODO for myself, I need to come back and review this commit for real if it really is important, since I haven't yet.

Reply:
> Do you understand the motivation for this commit

No, just seems like a cleanup to consolidate removing from clean and dirtylist in one method.

I'm not too familiar with the `FLUSHING` state but it looks like its not really checked much, maybe only as part of `isDirty()` calls. That said, I notice we aren't ever setting `flushStarted` to false after its been set to true (only on instantiation of an `Entry`). Not clear yet if that's intentional, though I suspect it's not since we'd previously move from `FLUSHING` to `CLEAN/NOT_IN_CACHE`. It may be the case that it's worth keeping the `FLUSHING` state, although I know that uncached `get()`s are set to `DIRTY` and put in the `dirtyList` now, so it might be tricky. Will look into it.

Reply:
Hm, removing `FLUSHING` from the list of states but adding a new `flushStarted` bool that has to be checked separately from the state doesn't really feel like a simplification, but I guess I should reserve judgment until I see how it's used by the later commits.

Although after reading through this commit in its entirety, I'm pretty suspicious of the risks that it introduces given that the benefit isn't obvious yet. And regarding the benefits, we still reference "flushing" entries in a bunch of places, it's just that now it's lower-case "flushing" rather than an explicit FLUSHING state.

Reply:
So I tried the naive approach of switching back to `FLUSHING` and dropping `flushStarted`, and ran into issues with the **ActorCache deleteAll() during transaction commit** test. Specifically, following `put(bar, 456)` we do a `deleteAll()`, and that `deleteAll()` is registered before we get around to flushing that `put(bar, 456)`. This means when we get around to attempting the `startFlushTransaction()`, we have already `requestedDeleteAll`.

In fact, the prior flush hasn't even finished yet so [we modify](https://github.com/cloudflare/workerd/blob/9651cc9b814b20d4674f1b0cba2e83ebbfbc8c38/src/workerd/io/actor-cache.c%2B%2B#L2484-L2496) our `deletedDirty` collection (so it only contains `bar`), which ensures we'll flush that `put(bar)` before doing the `deleteAll()`.

This all kind of breaks when we switch back to having a `FLUSHING` state, since `deleteAll()` calls `ActorCache::clear()`, which calls `removeEntry()` on all cached entries in the map, and `removeEntry()` (now) sets the `EntrySyncStatus` to `NOT_IN_CACHE`. Since we're no longer tracking the entry as waiting to be flushed (via `flushStarted`), we assume it has already been flushed and the `put()` is abandoned in favour of the `deleteAll()`.

This should be fixable, although it might make tracking `NOT_IN_CACHE` a bit less clear.

Reply:
We can go either way on this. Stylistically it doesn't seem like a clear improvement, but it's not a fundamentally bad approach. And it comes with some risks (as called out in my other comments on the commit), but if it simplifies the later changes in this PR then that's a solid point in its favor.

At this point I see it mostly as a matter of picking which battle to take on -- either making sure the things I commented on aren't actually an issue (and/or more surgically undoing the parts that are) or figuring out how to rip out this commit from the stack of changes.

> and removeEntry() (now) sets the EntrySyncStatus to NOT_IN_CACHE

It doesn't if we were to just rip out the whole commit, but I'm assuming that's difficult due to how later commits build on it?